File: Java/162.java
class Solution {
    public int findPeakElement(int[] nums) {
        int mid = 0, left = 0, right = nums.length - 1;
        while(left <= right){
            mid = left + (right - left)/2;
            try{
                if(nums[mid - 1] > nums[mid]){
                    right = mid - 1;
                    continue;
                }
            }
            catch(Exception e){
            }
            try{
                if(nums[mid + 1] > nums[mid]){
                    left = mid + 1;
                    continue;
                }
            }
            catch(Exception e){
            }
            return mid;
        }
        return mid;
    }
}

File: Python/162.py
class Solution(object):
    def findPeakElement(self, nums):
        nums = [-float("inf")] + nums + [-float("inf")]
        left = 1
        right = len(nums) - 2
        while left <= right:
            mid = left + (right - left)//2
            if max(nums[mid-1:mid+2]) == nums[mid]:
                return mid - 1
            if nums[mid-1] > nums[mid]:
                right = mid - 1
            else:
                left = mid + 1